ReactAndroid/src/main/java/com/facebook/react/views/text/CustomLineHeightSpan.java

```diff
@@ -45,10 +45,9 @@ public void chooseHeight(
       // Show all ascent, descent, bottom, as much top as possible
       fm.top = fm.bottom - mHeight;
     } else {
-      // Show proportionally additional ascent and top
+      // Show proportionally additional top
       final int additional = mHeight - (-fm.top + fm.bottom);
       fm.top -= additional;
-      fm.ascent -= additional;
```

Comment:
@bartolkaruza & @andreicoman11:
I've been looking at this PR and noticed a possible regression. Line height seems to be broken on wrapped lines of text when line height is greater than font size. It seems to be related to this line of code.

Here is an example of what I'm seeing:
![screen shot 2017-11-03 at 11 50 19](https://user-images.githubusercontent.com/2070959/32372573-5ac9e352-c08d-11e7-9f74-61bb51d8f4af.png)

JSX:
```
<View style={{backgroundColor: "yellow", paddingTop: 100, paddingBottom: 100}}>
    <Text style={{fontSize: 20, lineHeight: 60, backgroundColor: "blue", color: "white"}}>
        Fontsize 20 Lineheight 60 qwertyuiopasdfghjklzxcvbnm QWERTYUIOPASDFGHJKLZXCVBNM
        Fontsize 20 Lineheight 60 qwertyuiopasdfghjklzxcvbnm QWERTYUIOPASDFGHJKLZXCVBNM
    </Text>
</View>```

Reply:
Restoring `fm.ascent -= additional;` gives me this behaviour:
![screen shot 2017-11-03 at 11 56 40](https://user-images.githubusercontent.com/2070959/32372796-4b3a2766-c08e-11e7-9ba1-f4263157fb9b.png)

Reply:
Proposed fix to make Android consistent with iOS:
```
      // Show proportionally additional ascent / top & descent / bottom
      final int additional = mHeight - (-fm.top + fm.bottom);

      fm.top -= additional / 2;
      fm.ascent -= additional / 2;
      fm.descent += additional / 2;
      fm.bottom += additional / 2;
```
Giving following behaviour:
![screen shot 2017-11-03 at 12 11 40](https://user-images.githubusercontent.com/2070959/32373248-4246d80a-c090-11e7-84d7-7b4f7c7954ad.png)

Reply:
@dickie81 Thank you for looking out, I did not check the full range of test cases on the actual UI when changing the style back to the original style, leaning on the unit tests instead. I'll try out your approach against the cases I have locally for all the lineHeight scenario's and post the test setup so everyone can verify.

Reply:
@dickie81 I've been testing the ui effects and there is definitely a regression in the case of bigger lineHeight. Unfortunately your proposed solution still causes cut-off at the bottom, when the lineHeight is small.
@andreicoman11 I'm looking into this more in depth, this PR is not ready yet as it stands now.

Here is the testing setup that you can use to verify on your end: https://gist.github.com/bartolkaruza/b74f228636bb78fea00aa8bd82e3766d

And these are the 4 different implementation compared to iOS. Not one is quite right yet.

Current PR state after review changes:
---
<img width="862" alt="new-style-bartolkaruza" src="https://user-images.githubusercontent.com/1605731/32405086-bfbcae12-c15e-11e7-9ad4-ef444aaae54a.png">

Original implementation before PR:
---
<img width="860" alt="original-andreicoman11" src="https://user-images.githubusercontent.com/1605731/32405087-bfd71450-c15e-11e7-8ec9-b9ebee5a1791.png">

Implementation before review changes:
---
<img width="861" alt="original-bartolkaruza" src="https://user-images.githubusercontent.com/1605731/32405088-c02e84ba-c15e-11e7-8b1c-fea6b5ac0e09.png">

Proposed change by @dickie81 :
---
<img width="855" alt="proposed-dickie81" src="https://user-images.githubusercontent.com/1605731/32405089-c043bf38-c15e-11e7-884e-95382b25d45f.png">